src/player/features/loop/ui/LoopButton.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+import React from "react";
+
+import { observer } from "mobx-react-lite";
+import { PiRepeat } from "react-icons/pi";
+
+import { usePlayer } from "@/player/entities/player";
+import { IconButton } from "@/shared/ui/inputs";
+import { LoopFeature } from "../model/LoopFeature";
+
+export const LoopButton = observer(function LoopButton() {
+  const player = usePlayer();
+  const [loopFeature] = React.useState(() => new LoopFeature({ player }));
+  const [isRepeating, setIsRepeating] = React.useState(false);
+
+  const handleClick = () => {
+    setIsRepeating((prevState) => {
+      const newRepeatingState = !prevState;
+      loopFeature.execute(newRepeatingState);
+
+      return newRepeatingState;
+    });
+  };
+
+  return (
+    <IconButton tooltip="Repeat" size="small" onClick={handleClick}>
+      <PiRepeat
+        color={
+          isRepeating ? "var(--color-bright-accent)" : "var(--color-subdued)"
```

Comment:
And here we would have: `player.isLooping ? ... : ...`

By the way, do we have to use css custom properties here? Using tailwind classes doesn't work?

Reply:
Corrected !